worlds/terraria/__init__.py

```diff
@@ -0,0 +1,329 @@
+            ):
+                # Event
+                locations.append(rule)
+
+        item_count = 0
+        items = []
+        for rule, flags, _, _ in rules[:goal]:
+            if not self.calamity and "Calamity" in flags:
+                continue
+            if "Item" in flags:
+                # Item
+                item_count += 1
+                if rule not in goal_locations:
+                    items.append(rule)
+            elif (
+                "Achievement" not in flags
+                and "Location" not in flags
+                and "Item" not in flags
+            ):
+                # Event
+                items.append(rule)
+
+        extra_checks = self.multiworld.fill_extra_checks_with[self.player].value
+        ordered_rewards = [
+            reward
+            for reward in labels["ordered"]
+            if self.calamity or "Calamity" not in rewards[reward]
+        ]
+        while extra_checks == 1 and item_count < location_count and ordered_rewards:
+            items.append(ordered_rewards.pop(0))
+            item_count += 1
+
+        random_rewards = [
+            reward
+            for reward in labels["random"]
+            if self.calamity or "Calamity" not in rewards[reward]
+        ]
+        self.multiworld.random.shuffle(random_rewards)
+        while extra_checks == 1 and item_count < location_count and random_rewards:
+            items.append(random_rewards.pop(0))
+            item_count += 1
+
+        while item_count < location_count:
+            items.append("Reward: Coins")
+            item_count += 1
+
+        self.ter_items = items
+        self.ter_locations = locations
+
+        self.ter_goals = ter_goals
+        self.goal_items = goal_items
+        self.goal_locations = goal_locations
+
+    def create_regions(self) -> None:
+        menu = Region("Menu", self.player, self.multiworld)
+
+        for location in self.ter_locations:
+            menu.locations.append(
+                TerrariaLocation(
+                    self.player, location, location_name_to_id.get(location), menu
+                )
+            )
+
+        self.multiworld.regions.append(menu)
+
+    def create_item(self, item: str) -> TerrariaItem:
+        if item in progression:
+            classification = ItemClassification.progression
+        else:
+            classification = ItemClassification.filler
+
+        return TerrariaItem(item, classification, item_name_to_id[item], self.player)
+
+    def create_items(self) -> None:
+        for item in self.ter_items:
+            if (rule_index := rule_indices.get(item)) is not None:
+                _, flags, _, _ = rules[rule_index]
+                if "Item" in flags:
+                    name = flags.get("Item") or f"Post-{item}"
+                else:
+                    continue
+            else:
+                name = item
+
+            self.multiworld.itempool.append(self.create_item(name))
+
+        self.locked_items = {}
```

Comment:
Define this field in your class. It's bad practice to create new instance (or class) fields outside of `__init__`.

```python
class TerrariaWorld(World):
    locked_items: dict  # Or whatever specific Dict[_, _] type this is.
```